src/nupic/algorithms/Cells4.cpp

```diff
@@ -2827,42 +2787,15 @@ void Cells4::computeForwardPropagation(CState& state)
 {
   // Zero out previous values
   // Using memset is quite a bit faster on laptops, but has almost no effect
-  // on Neo15!
+  // on Neo15! // TODO shouldn't we optimize for generic (laptop) CPUs rather than
+  // a cellphone CPU
   _inferActivity.reset();
 
   // Compute cell and segment activity by following forward propagation
   // links from each source cell.  _cellActivity will be set to the total
   // activity coming into a cell.
-#ifdef NTA_PLATFORM_darwin86
-  const UInt multipleOf8 = 8 * (_nCells/8);
-  UInt i;
-  for (i = 0; i < multipleOf8; i += 8) {
-    UInt64 eightStates = * (UInt64 *)(state.arrayPtr() + i);
-    for (int k = 0; eightStates != 0  &&  k < 8; eightStates >>= 8, k++) {
-      if ((eightStates & 0xff) != 0) {
-        std::vector< OutSynapse >& os = _outSynapses[i + k];
-        for (UInt j = 0; j != os.size(); ++j) {
-          UInt dstCellIdx = os[j].dstCellIdx();
-          UInt dstSegIdx = os[j].dstSegIdx();
-          _inferActivity.increment(dstCellIdx, dstSegIdx);
-        }
-      }
-    }
-  }
-
-  // process the tail if (_nCells % 8) != 0
-  for (i = multipleOf8; i < _nCells; i++) {
-    if (state.isSet(i)) {
-      std::vector< OutSynapse >& os = _outSynapses[i];
-      for (UInt j = 0; j != os.size(); ++j) {
-        UInt dstCellIdx = os[j].dstCellIdx();
-        UInt dstSegIdx = os[j].dstSegIdx();
-        _inferActivity.increment(dstCellIdx, dstSegIdx);
-      }
-    }
-  }
-#else
-  const UInt multipleOf4 = 4 * (_nCells/4);
+  //TODO above was multipleOf8 for NON-darwin, here is multipleOf4 !
+  const UInt multipleOf4 = 4 * (_nCells/4); //TODO remove for small penalty, see "tail"
```

Comment:
that was my concern too. 26affc9 should fix it. It worked before because 64bit could work with the 32bit, and noone is using the darwin86 arch